### ISSUE TITLE
`reportMissingSuperCall` false negative when the super method is conditionally called

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#823]
Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=MYGwhgzhAEBiD28BcBYAUNT0AmBTAZtAPpECWAdqQC4kAUEuI%2BAlNALQB80AcvObqgxZhAB0gR06UOOgAhMACdaCeM0HC8hEhWp0GTVpx58B6YcNKEARohC01Z804gBXEbiXMAdNso0i9kA)

```python
class Foo:
    def __init__(self) -> None:
        pass

class Bar(Foo):
    def __init__(self) -> None: # no error
        if bool():
            super().__init__()
```
